x-pack/platform/plugins/shared/onechat/server/utils/kibana_mcp_http_transport.ts

```diff
@@ -0,0 +1,377 @@
+  private _logger: Logger;
+  private _responseCallbacks: Map<string, (response: IKibanaResponse) => void> = new Map();
+
+  sessionId?: string | undefined;
+  onclose?: () => void;
+  onerror?: (error: Error) => void;
+  onmessage?: (message: JSONRPCMessage, extra?: { authInfo?: AuthInfo }) => void;
+
+  constructor(options: { sessionIdGenerator?: () => string; logger: Logger }) {
+    this.sessionIdGenerator = options.sessionIdGenerator;
+    this._logger = options.logger;
+  }
+
+  /**
+   * Registers a callback for when a response is ready for a specific stream
+   */
+  private registerResponseCallback(
+    streamId: string,
+    callback: (response: IKibanaResponse) => void
+  ) {
+    this._responseCallbacks.set(streamId, callback);
+  }
+
+  /**
+   * Handles an incoming HTTP request, whether GET or POST
+   */
+  async handleRequest(req: KibanaRequest, res: KibanaResponseFactory): Promise<IKibanaResponse> {
+    if (req.route.method === 'post') {
+      return await this.handlePostRequest(req, res);
+    } else {
+      return await this.handleUnsupportedRequest(res);
+    }
+  }
+
+  /**
+   * Starts the transport. This is required by the Transport interface but is a no-op
+   * for the Streamable HTTP transport as connections are managed per-request.
+   */
+  async start(): Promise<void> {
+    if (this._started) {
+      throw new Error('Transport already started');
+    }
+    this._started = true;
+  }
+
+  /**
+   * Closes the transport and cleans up resources
+   */
+  async close(): Promise<void> {
+    this._streamMapping.clear();
+
+    // Clear any pending responses
+    this._requestResponseMap.clear();
+    this._requestToStreamMapping.clear();
+    this._responseCallbacks.clear();
+
+    // Clear session state
+    this.sessionId = undefined;
+    this._initialized = false;
+    this.onclose?.();
+  }
+
+  /**
+   * Handles POST requests containing JSON-RPC messages compatible with MCP specification
+   */
+  async handlePostRequest(
+    request: KibanaRequest<unknown, unknown, unknown>,
+    responseFactory: KibanaResponseFactory
+  ): Promise<IKibanaResponse> {
+    try {
+      this._logger.debug('Processing POST request');
+
+      // Validate the Accept header
+      const acceptHeader = request.headers.accept;
+
+      // The client MUST include an Accept header, listing application/json as supported content type.
+      if (!acceptHeader?.includes('application/json')) {
+        this._logger.warn('Request rejected: Invalid Accept header');
+        return responseFactory.customError({
+          statusCode: 406,
+          body: JSON.stringify({
+            jsonrpc: '2.0',
+            error: {
+              code: CONNECTION_CLOSED,
+              message: 'Not Acceptable: Client must accept application/json',
+            },
+            id: null,
+          }),
+        });
+      }
+
+      const ct = request.headers['content-type'];
+
+      if (!ct || !ct.includes('application/json')) {
```

Comment:
NIT: `request.headers`'s values can be `string` or `string[]`

https://github.com/elastic/kibana/blob/6049493e4a2372ea22986b231ad2bd59584fe9b8/src/core/packages/http/server/src/router/headers.ts#L50-L52

Wondering if we are fine using an `includes` check either way? Maybe for arrays we want to check `includes` on each element instead (e.g. `content-type : ['application/json; charset=utf-8']` would fail the check atm)